packages/core/src/geometry/GeometrySystem.js

```diff
@@ -586,22 +586,34 @@ export default class GeometrySystem extends System
      */
     draw(type, size, start, instanceCount)
     {
-        const { gl } = this;
+        const { gl, renderer: { context } } = this;
         const geometry = this._activeGeometry;
 
         // TODO.. this should not change so maybe cache the function?
 
         if (geometry.indexBuffer)
         {
-            if (geometry.instanced)
+            const byteSize = geometry.indexBuffer.data.BYTES_PER_ELEMENT;
+            const glType = byteSize === 2 ? gl.UNSIGNED_SHORT : gl.UNSIGNED_INT;
+
+            if (byteSize === 2 || context.webGLVersion === 2 || context.extensions.uintElementIndex)
```

Comment:
Could add this line at the top when the context is created:
```this.canUseUInt32 = context.webGLVersion === 2 || context.extensions.uintElementIndex;```

Then this line could become:
```if(byteSize === 2 || (this.canUseUInt32 && byteSize === 4))```